### DISCUSSION TITLE
Hydrus on Linux

### DISCUSSION THREAD

**Discussion by parisaheidary** [phydrus/phydrus#51] [Compiling Fortran Code] [answered]
Hello everyone,
I am trying to run Hydrus on Linux. I founded this link, I know it is a python package, but I am wondering if anyone has any experience in running Hydrus on Linux. I would really appreciate any help.

Thanks,
Parisa

**Comment by martinvonk** [comment, marked as answer]
Hi Parisa,

You can not run the HYDRUS-1D GUI (or 2D/3D) on Linux (or MacOS). You would need a programm like Wine (https://www.winehq.org/) or a virtual machine/emulator to do this. Would not recommend this if there is another option...

So it you only need HYDRUS-1D the other option is to compile the source code (phydrus/source_code) yourself. You can do this relatively easily if you are in Linux with gfortran. After you compiled the code into an executables you can use Phydrus to create input files, run the executable and visualise results. This should work as I have done it myself.

Hope this helps and feel free to ask if you have another question.

Kind regards,
Martin

**Comment by parisaheidary** [comment]
Thank you so much for your reply. Actually, I could compile the source code in Linux with gfortran (using: https://github.com/AgriHarmony/HYDRUS-1-D-gfortran). It is running perfectly for some examples that are provided, but not for all of them. And also I am getting errors once I provide METEO.IN file as the inputs of the Hydrus. Any help in this regard is really appreciated.

Also, can I create my meteo.in using phydrus? because I need that file as an input to calculate ET.

**Reply by mvremec** [reply]
Hi Parisa, 

We did not include any functions to provide METEO.in in Phydrus, as we encourage the use of external libraries to estimate evapotranspiration and then use the Ritchie (1972) formula to partition ET to soil evaporation (rSoil) and transpiration (rRoot).

Check [example 3](https://github.com/phydrus/phydrus/blob/master/examples/hydrus_orig/Example_3%20-%20hydrus%20tutorial.ipynb) on how to include atmospheric boundary conditions (precipitation, soil evaporation, transpiration,...) in Phydrus.
Have a look at [PyEt](https://github.com/phydrus/pyet) to estimate Potential Evapotranspiration.

Also, check these two discussions #48 and #28.
Best,
Matevz

**Reply by parisaheidary** [reply]
Thank you so much for your reply. I am a little bit confused, so how the actual ET will be considered? if I provide the potential ET (rSoil and rRoot potential) as my input, does the code calculate the actual Et itself? And may I know how?

I really appreciate your help.

Regards,
Parisa

**Comment by parisaheidary** [comment]
Hello again,
I did the same thing that you mentioend, I modified my inputs and calculate ET (then rSoil and rRoot) using that library. but still, I am getting an error of floating point- erroneous arithmetic operation.  Can you please help me to figure it out, is it something with my compiled code? 
Thank you so much.

**Reply by martinvonk** [reply]
Hi Parisa,

If I understand it correctly you compiled the source code from AgriHarmony/HYDRUS-1-D-gfortran. To make sure Phydrus runs smoothly, try compiling the source code from phydrus/source_code (https://github.com/phydrus/source_code). That should I think. Otherwise please share your Python code here so I can take a look.

**Reply by parisaheidary** [reply]
Thank you so much. I will do that and let you know.
just to make sure, which one should I use? Source or Source_mcmc?

Regards,
Parisa

**Reply by martinvonk** [reply]
I only use the Source one, not the Source_mcmc. But in the end, both should be fine I think. Nevertheless, just start with Source.

**Reply by parisaheidary** [reply]
That worked!!!! 
Thank you very much, I really appreciate your help. I have been struggling with this problem for several weeks.

**Reply by martinvonk** [reply]
Good to hear! Good luck with your work and feel free to ask if more problems occur.
